app/testOnly/controllers/SessionStorageServiceController.scala

```diff
@@ -0,0 +1,91 @@
+                                                implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                                val authorisedFunctions: FrontendAuthorisedFunctions,
+                                                val auth: AuthenticatorPredicate,
+                                                val sessionDataService: SessionDataService
+                                               ) extends ClientConfirmedController {
+
+  def show(): Action[AnyContent] = auth.authenticatedAction(isAgent = false) {
+    implicit user =>
+      handleShow(isAgent = false)
+  }
+
+  def showAgent: Action[AnyContent] = auth.authenticatedAction(isAgent = true) {
+    implicit mtdItUser =>
+      handleShow(isAgent = true)
+  }
+
+  private def handleShow(isAgent: Boolean)(implicit hc: HeaderCarrier, ec: ExecutionContext, user: MtdItUser[_]): Future[Result] = {
+    hc.sessionId match {
+      case Some(sessionId: SessionId) => post(isAgent = isAgent, sessionId) flatMap {
+        case Left(ex) =>
+          Logger("application").error(s"[SessionStorageServiceController][handleShow] " +
+            s"${if (isAgent) "Agent" else "Individual"} - POST user data to income-tax-session-data unsuccessful: - ${ex.getMessage} - ${ex.getCause} - ")
+          Future.successful(handleError(isAgent))
+        case Right(id: String) => sessionDataService.getSessionData(id) map {
+          case Left(ex) =>
+            Logger("application").error(s"[SessionStorageServiceController][handleShow] ${if (isAgent) "Agent" else "Individual"}" +
+              s" - GET user data request to income-tax-session-data unsuccessful: - ${ex.getMessage} - ${ex.getCause} - ")
+            handleError(isAgent)
+          case Right(model: SessionDataModel) =>
+            Ok(
+                s"User model:          ${model.toString}\n" +
+                s"session id:          ${model.sessionID}\n" +
+                s"mtditid:             ${model.mtditid}\n" +
+                s"nino:                ${model.nino}\n" +
+                s"saUtr:               ${model.saUtr}\n" +
+                s"client first name:   ${model.clientFirstName}\n" +
+                s"client last name:    ${model.clientLastName}\n" +
+                s"user type:           ${model.userType}\n")
+        }
+      }
+      case None =>
+        Logger("application").error(s"[SessionStorageServiceController][handleShow] ${if (isAgent) "Agent" else "Individual"}" +
+          s" - HeaderCarrier contained no sessionId!")
+        Future.successful(handleError(isAgent))
+    }
+  }
+
+  def post(isAgent: Boolean, sessionId: SessionId)(implicit hc: HeaderCarrier, ec: ExecutionContext, user: MtdItUser[_])
+  : Future[Either[Throwable, String]] =
+    sessionDataService.postSessionData(isAgent, sessionId)
+
+  private def handleError(isAgent: Boolean)(implicit request: Request[_]): Result = {
+    val errorHandler = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
```

Comment:
why not to show that there is no data ? we don't have to crash execution here

Reply:
for which scenario do you mean?

Reply:
as I understand current logic: you crash execution if there is no data ? so instead you can just show text on the page that there is no data

Reply:
in the current logic we post the data in the show method, so if there is no data then this is a big problem. Maybe I could show text on the page explaining this? E.g: No data in database for this session. Something has gone wrong.

Reply:
this is not decided yet how client would handle missing data, so I guess not critical at the moment